src/stores/AppStore/AppStore.ts

```diff
@@ -2108,6 +2111,34 @@ export class AppStore {
         autorun(() => {
             this.activateStatsPanel(this.preferenceStore.statsPanelEnabled);
         });
+
+        // listen devicePixelRatio
+        let remove = null;
+        const updatePixelRatio = () => {
+            if (remove != null) {
+                remove();
+            }
+            const mqString = `(resolution: ${window.devicePixelRatio}dppx)`;
+            const media = matchMedia(mqString);
+            media.addEventListener("change", updatePixelRatio);
+            remove = () => {
+                media.removeEventListener("change", updatePixelRatio);
+            };
+
+            this.pixelRatio.previous = this.pixelRatio.current;
+            this.pixelRatio.current = devicePixelRatio;
+            this.handleInvariantImageSize();
```

Comment:
Wonder if we could set `window.devicePixelRatio` and (unupdated) `pixelRatio.current` as inputs for `handleInvariantImageSize`, so that we don't need to store `pixeRatio.previous`.